src/ACEtk/block/details/ArrayData/src/ctor.hpp

```diff
@@ -0,0 +1,95 @@
+private:
+
+template < typename Number >
+ArrayData( std::string&& name, std::vector< std::vector< Number > >&& values,
+           unsigned int n, unsigned int m ) :
+  Base( std::move( name ), generateXSS( std::move( values ) ) ),
+  nelements_( n ), narrays_( m ) {}
+
+protected:
+
+ArrayData() = default;
+
+ArrayData( const ArrayData& ) = default;
+ArrayData( ArrayData&& ) = default;
+
+/**
+ *  @brief Convenience constructor
+ *
+ *  @param[in] name     the name of the block
+ *  @param[in] values   the values
+ */
+template < typename Number >
+ArrayData( std::string&& name, std::vector< std::vector< Number > >&& values ) :
+  ArrayData( std::move( name ), std::move( values ), values.front().size(),
+             values.size() ) {}
+
+/**
+ *  @brief Convenience constructor
+ *
+ *  @param[in] name     the name of the block
+ *  @param[in] array1   the values for array 1
+ *  @param[in] array2   the values for array 2
+ *  @param[in] array3   the values for array 3
+ *  @param[in] array4   the values for array 4
+ *  @param[in] array5   the values for array 5
+ */
+template < typename Number >
+ArrayData( std::string&& name,
+           std::vector< Number >&& array1,
+           std::vector< Number >&& array2,
+           std::vector< Number >&& array3,
+           std::vector< Number >&& array4,
+           std::vector< Number >&& array5 ) :
+  ArrayData( std::move( name ),
+             std::vector< std::vector< Number > >{
```

Comment:
I actually thought of that, but didn't want to put in the time to make it work since it is a base class and as such an implementation detail that no one will interact with directly.